db/seeds.rb

```diff
@@ -1,16 +1,56 @@
-# Создадим записи в базе данных для проверки,
-# восклицательный знак после create стоит для вызова исключения.
+# encoding: utf-8
+require "open-uri"
+require "nokogiri"
 
-Card.create!([
-  { original_text: "Привет. Как дела?",
-    translated_text: "Hello. How are you?",
-    review_date: Date.current },
 
-  { original_text: "Откуда ты?",
-    translated_text: "where are you from?",
-    review_date: Date.yesterday },
+class WordsDownloader
+  def initialize(url, original_text_col = 2, translated_text_col = 3)
+    @url = url
+    @original_text_col = original_text_col
+    @translated_text_col = translated_text_col
+  end
+
+  def get_words(&closure)
+    get_rows if @rows == nil
+    words = []
+    closure = if closure.nil?
+      lambda { |row| return row.css("td") }
+    else
+      lambda { closure }
+    end
+    @rows.each do |row|
+      cols = closure.call(row)
+      words << [cols[@original_text_col].text, cols[@translated_text_col].text]
+    end
+    words
+  end
+
+  def get_rows(&closure)
+    doc = Nokogiri::HTML(open(@url))
+    closure = if closure.nil?
+      lambda { return doc.css("table.topwords").css("tr")[1..-1] }
+    else
+      lambda { closure }
+    end
+    @rows = closure.call
+  end
+end
+
+class CardCreator < WordsDownloader
+  def create
+    words = get_words
+    words.each do |original_text, translated_text|
+      Card.create!({ 
+        original_text: original_text,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.